src/BcPrebidVast.js

```diff
@@ -0,0 +1,365 @@
+var _prebidGlobal = require('./PrebidGlobal.js');
+var _vastManager = require('./VastManager.js');
+var _prebidCommunicator = require('./PrebidCommunicator.js');
+var _logger = require('./Logging.js');
+var _prefix = 'PrebidVast->';
+
+var $$PREBID_GLOBAL$$ = _prebidGlobal.getGlobal();
+
+_logger.always(_prefix, 'Version 0.0.10');
+
+var BC_prebid_in_progress = $$PREBID_GLOBAL$$.plugin_prebid_options && $$PREBID_GLOBAL$$.plugin_prebid_options.biddersSpec;
+
+// the function does bidding and returns bids thru callback
+function doPrebid(options, callback) {
+	if ($$PREBID_GLOBAL$$.bc_pbjs && options.biddersSpec) {
+		  var pbjs = $$PREBID_GLOBAL$$.bc_pbjs || {};
+	      pbjs.que = pbjs.que || [];
+
+	      //
+	      // Prebid Video adUnit
+	      //
+	      var logBids = function(bids) {
+	    	  _logger.log(_prefix, 'MESSAGE: got bids back: ', bids);
+	      };
+
+	      pbjs.que.push(function() {
+	        pbjs.addAdUnits(options.biddersSpec); // add your ad units to the bid request
+
+	        if (options.prebidConfigOptions) {
+	        	// DFP reqired prebid cache
+			    if (!options.enablePrebidCache && options.prebidConfigOptions.cache && !options.dfpParameters) {
+			    	delete options.prebidConfigOptions.cache;
+			    }
+	        	pbjs.setConfig(options.prebidConfigOptions);
+	        }
+
+	        // activate prebid cache (DFP reqired prebid cache)
+		    if (options.enablePrebidCache || options.dfpParameters) {
+			    pbjs.setConfig({
+			        usePrebidCache: true
+			    });
+		    }
+
+	        pbjs.requestBids({
+	          timeout: (options.prebidTimeout && options.prebidTimeout > 0) ? options.prebidTimeout : 700,
+	          bidsBackHandler: function(bids) { // this function will be called once bids are returned
+	            logBids(bids);
+	            callback(bids);
+	          }
+	        });
+		  });
+	}
+	else {
+		callback(null);
+	}
+}
+
+function dispatchPrebidDoneEvent() {
+	var event;
+	if (typeof Event === 'function') {
+		event = new Event('prebid_done_loading_script');
+	}
+	else {
+		event = document.createEvent('Event');
+		event.initEvent('prebid_done_loading_script', true, true);
+	}
+	document.dispatchEvent(event);
+}
+
+// the function loads prebid.js and does header bidding if needed
+function loadPrebidScript(options, fromHeader) {
+    var pbjsScr = document.createElement('script');
+    pbjsScr.onload = function() {
+    	$$PREBID_GLOBAL$$.bc_pbjs = pbjs;
+    	// do header bidding if needed
+    	if (fromHeader && options && options.biddersSpec) {
+    		BC_prebid_in_progress = true;
+    		doPrebid(options, function(bids) {
+    			var arrBids = (bids && bids[options.biddersSpec.code]) ? bids[options.biddersSpec.code].bids : [];
+    			_logger.log(_prefix, 'bids for bidding: ', arrBids);
+    			if (arrBids && Array.isArray(arrBids)) {
+	    			if (options.dfpParameters) {
+	    				// use DFP server if DFP parameters present in options
+						_logger.log(_prefix, 'Use DFP');
```

Comment:
deeply nested code is more difficult to debug and test. It's better to break into smaller functions.

Reply:
I don't think it is deeply nested code. Also it is not a long code (about 40 lines) and it is a very strait forward code.

Reply:
it's nested 7 indentations in so I'd have to disagree with you there.